packages/tizen_bundle/lib/tizen_bundle.dart

```diff
@@ -0,0 +1,333 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:tizen_interop/4.0/tizen.dart';
+import 'package:tizen_log/tizen_log.dart';
+
+/// The class for the property type of the bundle.
+class _BundleTypeProperty {
+  /// The array type.
+  static const int array = 0x0100;
+
+  /// The measurable type.
+  static const int measurable = 0x0400;
+}
+
+/// The class for the type of the bundle.
+enum BundleType {
+  /// The none type.
+  none(0),
+
+  /// The string type.
+  string(1 | _BundleTypeProperty.measurable),
+
+  /// The string list Type.
+  strings(1 | _BundleTypeProperty.measurable | _BundleTypeProperty.array),
+
+  /// The bytes type.
+  bytes(2);
+
+  /// Creates an instance of [BundleType] with the given argument.
+  const BundleType(this.value);
+
+  /// The value of the bundle type.
+  final int value;
+}
+
+/// The class for information of the key of the bundle object.
+class KeyInfo {
+  /// Creates an instance of [KeyInfo] with the given arguments.
+  KeyInfo(this.name, this.type);
+
+  /// The name of the key.
+  final String name;
+
+  /// The type of the key.
+  final BundleType type;
+
+  /// Check whether the type of the key is array or not.
+  bool get isArray => type.value & _BundleTypeProperty.array as bool;
+}
+
+class _BundleErrorFactory {
+  factory _BundleErrorFactory() {
+    return _instance;
+  }
+
+  _BundleErrorFactory._internal();
+
+  static final _BundleErrorFactory _instance = _BundleErrorFactory._internal();
+
+  final Map<int, String> _errorMessages = {
+    bundle_error_e.BUNDLE_ERROR_OUT_OF_MEMORY: 'Out of memory',
+    bundle_error_e.BUNDLE_ERROR_INVALID_PARAMETER: 'Invalid parameter',
+    bundle_error_e.BUNDLE_ERROR_KEY_EXISTS: 'Key already exists',
+    bundle_error_e.BUNDLE_ERROR_KEY_NOT_AVAILABLE: 'Key does not exists',
+  };
+
+  void throwException(int error) {
+    if (_errorMessages.containsKey(error)) {
+      throw Exception(_errorMessages[error]);
+    } else {
+      throw Exception('Unknown error: $error');
+    }
+  }
+}
+
+/// A bundle object represents a bundle.
+/// A bundle holds items (key-value pairs) and can be used with other Tizen APIs.
+/// Keys can be used to access values.
+/// This class is accessed by using a constructor to create a new instance of this object.
+/// A bundle instance is not guaranteed to be thread safe if the instance is modified by multiple threads.
+class Bundle {
+  /// Creates an instance of [Bundle].
+  Bundle() : _handle = tizen.bundle_create() {
+    _finalizer.attach(this, this, detach: this);
+  }
+
+  /// Creates an instance of [Bundle] with the given raw data.
+  Bundle.fromRaw(String raw)
+      : _handle =
+            tizen.bundle_decode(raw.toNativeInt8().cast<Uint8>(), raw.length) {
```

Comment:
Note that you can declare `_handle` as a `late final` and initialize its value in  the constructor body.

Reply:
Done.